### PR TITLE
Prepare code for development

### PR DESCRIPTION
**Pull request by javierav**
This PR prepares the code for the planned developments:

* Add rubocop with standard gem to ensure code consistency and style, also running it in the CI.
* Add a Gemfile for development dependencies, a best practice.
* Add the development gems' [binstubs](https://stackoverflow.com/questions/45122050/what-does-bundle-install-binstubs-serve-for-us) to the `bin` folder to avoid having to prefix commands with `bundle exec`.
* Adds support for [direnv](https://direnv.net), a utility that allows, among other things, to automatically add the `bin` folder to the `PATH` environment variable every time we enter the project directory, closely related to binstubs. This allows us to directly use the `rake` command in the console and actually execute `bin/rake`, which in turn is like writing `bundle exec rake`.
* Adds an IRB console in `bin/console` with the preloaded library for testing in development.
* Set the gem to require ruby ​​>= 3.0
* Fix all rubocop warnings.
* Cleanup gitignore file.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -7,3 +7,4 @@ insert_final_newline = true
 indent_style = space
 indent_size = 2
 trim_trailing_whitespace = true
+max_line_length = 120
```
Comment (adrianthedev) [inline .editorconfig:10]: We don't really use that and it seems a bit arbitrary to add it for everyone.
We try to enforce code styling as little as possible.

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+require:
```
Comment (adrianthedev) [inline .rubocop.yml:1]: Can we use standard for this?
Again, we'd like to do as little yak-shaving as possible and standard has pretty good defaults.

**Review by adrianthedev — CHANGES_REQUESTED** [review]
This is awesome!
A lot of improvements here!

There's one thing I'd like us to be a little bit loose on.
Let's use `standard` as it provides a lot of good defaults and prevents us and future commiters to do a lot of yak shaving and fiddle with the linting rules.

Thanks a lot @javierav

**Comment by javierav** [comment]
@adrianthedev changes ready!

**Pull request merged**